Patches/CheckGameEndPatch.cs

```diff
@@ -337,7 +344,11 @@ public static bool Prefix()
                             break;
                     }
                 }
-
+                foreach (var pc in Main.AllPlayerControls.Where(x => x.IsNeutralApocalypse() && Main.AllAlivePlayerControls.All(p => p.IsNeutralApocalypse())))
```

Comment:
I think this code will better
`if (Main.AllPlayerControls.Count(x => x.IsNeutralApocalypse()) == Main.AllAlivePlayerControls.Length)`

Reply:
do you mean like this?
![image](https://github.com/user-attachments/assets/15612f1f-4da2-4dcd-8f43-0807f0c41a51)

Reply:
lmao, so when all 4 apocalypse member are dead, but 4 players are alive. They still win 😹

I think it's better to jsut move the all statement out of the where and put it in the if statement.

Reply:
wait i already have a CustomWinner case for apocalypse earlier in the code do I even need this

Reply:
ryuk help us oh god 😭

How the hell does this winner thing even cfunction

Reply:
idk man i didnt code it 😭

Reply:
ik....

Reply:
Damn I wrote the code wrong
It’s just strange that in Where constantly checks All

```
if (Main.AllAlivePlayerControls.All(p => p.IsNeutralApocalypse()))
{
    foreach (var pc in Main.AllPlayerControls.Where(x => x.IsNeutralApocalypse()))
    {
        if (!WinnerIds.Contains(pc.PlayerId))
            WinnerIds.Add(pc.PlayerId);
    }
}
```